message/job.py

```diff
@@ -0,0 +1,71 @@
+# ############################################################################### #
+# Autoreduction Repository : https://github.com/ISISScientificComputing/autoreduce
+#
+# Copyright &copy; 2020 ISIS Rutherford Appleton Laboratory UKRI
+# SPDX - License - Identifier: GPL-3.0-or-later
+# ############################################################################### #
+""" Represents the messages passed between AMQ queues"""
+import json
+
+
+# pylint:disable=too-many-instance-attributes
+class Message:
+    """
+    A class that represents an AMQ Message, these can be both serialised and un-serialised
+    for sending messages to and from AMQ
+    """
+
+    def __init__(self, serialised_object=None, overwrite=True):
+        """
+        If a serialised object is supplied then the class will be populated
+        :param serialised_object: a serialised json object (json.dumps)
+        :param overwrite: Should we overwrite existing variables?
+        """
+        self.description = None
+        self.facility = None
+        self.run_number = None
+        self.instrument = None
+        self.rb_number = None
+        self.started_by = None
+        self.file_path = None
+        self.overwrite = None
+        self.run_version = None
+        self.job_id = None
+        self.reduction_script = None
+        self.reduction_arguments = None
+        self.reduction_log = None
+        self.admin_log = None
+        self.return_message = None
+        self.retry_in = None
+        if serialised_object:
+            self.un_serialise(serialised_object, overwrite)
+
+    def to_dict(self):
+        """
+        Create a dictionary of all the class member variables
+        :return: member variables dictionary
+        """
+        return {key: value for key, value in self.__dict__.items()
+                if not key.startswith('__') and not callable(key)}
+
+    def serialise(self):
+        """
+        serialised member variables as a json dump
```

Comment:
Very minor: There's some inconsistency in whether capitals are used at the beginning of (i) descriptions (ii) param: (iii) return: